bundles/org.eclipse.swt/Eclipse SWT WebKit/gtk/org/eclipse/swt/browser/WebKit.java

```diff
@@ -784,13 +789,26 @@ public void create (Composite parent, int style) {
 				onResize (event);
 				break;
 			}
+			case SWT.KeyDown: {
+				if (event.keyCode == 'f' && (event.stateMask & SWT.CTRL) == SWT.CTRL) {
+					openSearchDialog();
+				}
+				break;
+			}
 		}
 	};
 	browser.addListener (SWT.Dispose, listener);
 	browser.addListener (SWT.FocusIn, listener);
 	browser.addListener (SWT.KeyDown, listener);
 	browser.addListener (SWT.Resize, listener);
 
+	browser.addDisposeListener(e -> {
+		closeSearchDialog();
+	});
+	browser.addControlListener(ControlListener.controlResizedAdapter(e -> closeSearchDialog()));
+	browser.addControlListener(ControlListener.controlMovedAdapter(e -> closeSearchDialog()));
+	browser.getShell().addControlListener(shellMoveListener);
```

Comment:
It would be nice if there is a way to close the search shell without needing to add this listener to the browsers shell. The move listener on the browser will get notifications only if the browser moves within the parent... Not if the whole window moves.

Since the search shell is opened at a specific location, it looks odd when moving the browsers shell - the search shell remains where it was opened.